tfjs-backend-webgpu/src/argminmax_webgpu.ts

```diff
@@ -67,6 +67,10 @@ export class ArgMinMaxProgram implements WebGPUProgram {
     const indexOutputCoords = (outputCoords: string, index: string) => {
       if (this.outputShape.length === 1) {
         return outputCoords;
+      } else if (this.outputShape.length === 5) {
```

Comment:
It's not good to use `getCoordFrom5D/6DIndex` which includes many `if-else` in the shader body.
How about having a helper function like below so that you can 1) avoid the if-else in the shader and 1) reuse the code in js, merge `indexOutputCoords` and `indexInputShape`:
```
const getCoordFromIndex = (coords: string, index: string, isScalar: boolean) =>
{
if (isScalar)
{
  return `${coords}`;
}
if (index === '0')
{
return `${outputCoords}.x`;
}
if (index === '1')
{
return `${outputCoords}.y`;
} else if (index === '2')
{
return `${outputCoords}.z`;
} else if (index === '3')
{
return `${outputCoords}.w`;
} else if (index === '4')
{
return `${outputCoords}.u`;
} else if (index === '5')
{
return `${outputCoords}.v`;
} 
}

Reply:
A second thought. Since i is a string, you can't directly compare with a string number. Maybe you need to change the string i to a number i. Then you can reuse your current helper function `getCoordsXYZ`.

Reply:
PTAL.